tests/D2L.CodeStyle.Analyzers.Test/Async/Generator/AsyncToSyncMethodTransformerTests.cs

```diff
@@ -121,21 +121,21 @@ internal sealed class AsyncToSyncMethodTransformerTests {
 	}
 
 	[Test]
-	public void CompletedTaskConfigureAwait() {
+	public void CompletedTaskConfigureAwaitFakeAsync() {
 		var actual = Transform( @"[GenerateSync] Task BarAsync() { return Task.CompletedTask; }" );
 
 		Assert.IsTrue( actual.Success );
 		Assert.IsEmpty( actual.Diagnostics );
-		Assert.AreEqual( "[Blocking] void Bar() { return ; }", actual.Value.ToFullString() );
+		Assert.AreEqual( "[Blocking] void Bar() { return; }", actual.Value.ToFullString() );
 	}
 
 	[Test]
 	public void FromResult() {
-		var actual = Transform( @"[GenerateSync] Task BarAsync() { return Task.FromResult( baz ); }" );
+		var actual = Transform( @"[GenerateSync] Task<Baz> BarAsync() { return Task.FromResult( baz ); }" );
 
 		Assert.IsTrue( actual.Success );
 		Assert.IsEmpty( actual.Diagnostics );
-		Assert.AreEqual( "[Blocking] void Bar() { return ( baz ); }", actual.Value.ToFullString() );
+		Assert.AreEqual( "[Blocking] Baz Bar() { return ( baz ); }", actual.Value.ToFullString() );

```

Comment:
I know this is existing, but was leaving the parentheses intentional?

Reply:
Not exactly intentional but seems like a small price to pay to keep 
`Transform( MemberAccessExpressionSyntax memberAccessExpr )`
relatively clean

Reply:
What's happening is that we are transforming `Task.FromResult` into `Parse( "" )` (which I think turns into an `IdentifierNameSyntax` with "missing token" placeholder) and the `(` and `)` are from the left over `ArgumentListSyntax`.

I think what we want to do is transform `InvocationExpression` and peek down into the `.Expression` to confirm that it is exactly `Task.FromResult` (if you do things funky like `var x = Task.FromResult` then its on you), and if so reach down into the `.Arguments[0]` and lift that up.

This has come up a couple of times now, I think we ought to have the tests validate that the generated parse tree is valid, or that ToString/Parse round trip in some reasonable way (like that we get an equivalent SyntaxTree from `Parse( output.ToString() )`.

It's not the end of the world that this is happening, but the (false) belief that we are generating valid parse trees gives me some peace of mind lol